### ISSUE TITLE
`object_length_linter` should detect generics

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#871]
As the `object_name_linter` already does, class suffixes for S3 generics shouldn't be part of the linted symbol.
That is, `object_length_linter(30)` (the default) shouldn't lint this example:

```
my_generic_function <- function(x, ...) {
  UseMethod("my_generic_function")
}

my_generic_function.my_cool_class <- function(x, ...) {
  cat("How cool and generic!\n")
}
```

**Comment by AshesITR** [comment]
For now, I've even disabled the linter alltogether in the `.lintr` config because it didn't habe any true positives in my project.

**Comment by MichaelChirico** [comment]
Should this be an option?

**Comment by MichaelChirico** [comment]
I think the new behavior is a happier default -- often it's out of your control if you're defining a method for someone else's class.

WDYT about a third option -- enforce the length on two parts (class name and generic name)? not sure it's worth the implementation & maintenance effort but it does make some sense

**Comment by AshesITR** [comment]
We could combine both by specifying two maximum lengths?
